server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java

```diff
@@ -243,8 +246,20 @@ public void putSynonymsSet(String resourceName, SynonymRule[] synonymsSet, Actio
             bulkRequestBuilder.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
                 .execute(deleteByQueryResponseListener.delegateFailure((bulkResponseListener, bulkResponse) -> {
                     if (bulkResponse.hasFailures() == false) {
-                        UpdateSynonymsResult result = created ? UpdateSynonymsResult.CREATED : UpdateSynonymsResult.UPDATED;
-                        bulkResponseListener.onResponse(result);
+                        UpdateSynonymsResultStatus updateSynonymsResultStatus = created
+                            ? UpdateSynonymsResultStatus.CREATED
+                            : UpdateSynonymsResultStatus.UPDATED;
+
+                        // auto-reload all reloadable analyzers (currently only those that use updateable synonym or keyword_marker filters)
+                        // TODO: reload only those analyzers that use this synonymsSet
```

Comment:
What are your thoughts for reloading just the analyzers for the specific syonyms set? Do it in a follow up PR? Or do we need some additional infra for retrieving the analyzers that use it?

Reply:
Right, we can do this in a follow-up. It is not trivial how to do this efficiently.